### DISCUSSION TITLE
what should i do if i want to forbid the child ForeignKey turn to null when i delete the parent

### DISCUSSION THREAD

**Discussion by anyfig** [sqlalchemy/sqlalchemy#9638] [Usage Questions] [answered]
```
class Child(db.Model):    
    __tablename__ = 'child'
    id = db.Column(db.Integer, autoincrement=True, primary_key=True)
    parent_id = db.Column(db.Integer,db.ForeignKey(f'parent.id', ondelete='NO ACTION'),nullable=True)

    parent = relationship('Parent',backref=db.backref('childs'))

class Parent(db.Model):    
    __tablename__ = 'parent'
    id = db.Column(db.Integer, autoincrement=True, primary_key=True)
```

when i db.session.delete(parent) ,
the related child's parent_id will turn to null.
i want to keep the parent_id what it used to be

**Comment by zzzeek** [comment, marked as answer]
relationship has an option [passive_deletes](https://docs.sqlalchemy.org/en/20/orm/relationship_api.html#sqlalchemy.orm.relationship.params.passive_deletes) which you would set to the value "all" to disable the "nulling out" of foreign keys.

a more long form discussion of passive_deletes is at https://docs.sqlalchemy.org/en/20/orm/cascades.html#passive-deletes .   the "all" option is mentioned in the "notes on passive deletes" section.

**Reply by anyfig** [reply]
thank you ，it works
